resolver/dns/dns_resolver.go

```diff
@@ -24,10 +24,29 @@
 package dns
 
 import (
+	"time"
+
 	"google.golang.org/grpc/internal/resolver/dns"
 	"google.golang.org/grpc/resolver"
 )
 
+// SetResolvingTimeout sets the maximum duration for DNS resolution requests.
+//
+// This function updates the global `ResolvingTimeout` variable, which specifies
+// how long a DNS resolution attempt will wait before it is canceled.
```

Comment:
Instead of mentioning "the global `ResolvingTimeout`", maybe something like "This function affects the global timeout used by all channels using the DNS name resolver scheme"?